src/extensions/extension-store.ts

```diff
@@ -0,0 +1,174 @@
+  private constructor() {
+    super({
+      configName: "lens-extension-store",
+      syncEnabled: false,
+    });
+  }
+
+  @observable version: ExtensionVersion = "0.0.0";
+  @observable extensions = observable.map<ExtensionId, LensExtension>();
+  @observable removed = observable.map<ExtensionId, LensExtension>();
+  @observable installed = observable.map<ExtensionId, InstalledExtension>([], { deep: false });
+
+  get folderPath(): string {
+    if (isDevelopment) {
+      return path.resolve(__static, "../src/extensions");
+    }
+    return "" // todo: figure out prod-path
+  }
+
+  async load() {
+    await this.loadExtensions();
+    return super.load();
+  }
+
+  autoEnableOnLoad(getLensRuntimeEnv: () => LensRuntimeRendererEnv, { delay = 0 } = {}) {
+    logger.info('[EXTENSIONS-STORE]: auto-activation loaded extensions: ON');
+    return reaction(() => this.installed.toJS(), installedExtensions => {
+      installedExtensions.forEach(({ extensionModule, manifest, manifestPath }) => {
+        let instance = this.getById(manifestPath);
+        if (!instance) {
+          const LensExtensionClass = extensionModule.default;
+          instance = new LensExtensionClass({ ...manifest, manifestPath, id: manifestPath }, manifest);
+          instance.enable(getLensRuntimeEnv());
+          this.extensions.set(manifestPath, instance); // save
+        }
+      })
+    }, {
+      fireImmediately: true,
+      delay: delay,
+    })
+  }
+
+  getExtensionByManifest(manifestPath: string): InstalledExtension {
+    let manifestJson: ExtensionManifest;
+    let mainJs: string;
+    try {
+      manifestJson = __non_webpack_require__(manifestPath); // "__non_webpack_require__" converts to native node's require()-call
+      mainJs = path.resolve(path.dirname(manifestPath), manifestJson.main);
+      mainJs = mainJs.replace(/\.ts$/i, ".js"); // todo: compile *.ts on the fly?
+      const extensionModule = __non_webpack_require__(mainJs);
+      return {
+        manifestPath: manifestPath,
+        manifest: manifestJson,
+        extensionModule: extensionModule,
+      }
+    } catch (err) {
+      console.error(`[EXTENSION-STORE]: can't load extension at ${manifestPath}: ${err}`, { manifestJson, mainJs });
+    }
+  }
+
+  @action
+  async loadExtensions() {
+    const extensions = await this.loadFromFolder(this.folderPath);
+    const extManifestMap = new Map(extensions.map(ext => [ext.manifestPath, ext]));
+    this.installed.replace(extManifestMap);
+  }
+
+  async loadFromFolder(folderPath: string): Promise<InstalledExtension[]> {
+    const paths = await fs.readdir(folderPath);
+    const manifestsLoading = paths.map(fileName => {
+      const absPath = path.resolve(folderPath, fileName);
+      const manifestPath = path.resolve(absPath, "package.json");
+      return fs.access(manifestPath, fs.constants.F_OK)
+        .then(() => this.getExtensionByManifest(manifestPath))
+        .catch(() => null)
```

Comment:
Yes, to extension developer it would be very beneficial information. I assume this is still a moving target, so I'll create one more todo item to epic issue about error handling.